### PR TITLE
Focus 2FA TOTP field

### LINKED ISSUE #9805: Automatically focus the 2FA input field
**What's the problem this feature will solve?**
When logging into PyPI with a 2FA-enabled account, the "authentication code" field should be auto-selected to simplify input by users.

**Describe the solution you'd like**
Modify the form design to put the focus on the primary field automatically.

**Additional context**
Tested under Linux with both Firefox (`78.4.1esr`) and Chromium (`86.0.4240.193`)

NB: If this feature interests the Warehouse project, I'll happily submit a pull-request to implement it.

### PR DESCRIPTION
**Pull request by ewjoachim**
Fixes #9805 

Tested succesfully:

Field is selected when there's 1 2FA method
Capture d’écran 2021-09-02 à 23 56 28

Field is not selected otherwise
Capture d’écran 2021-09-02 à 23 56 43

### CODE REVIEW

File: warehouse/templates/accounts/two-factor.html [hunk 1]
```diff
@@ -89,7 +89,17 @@ <h2>{% trans %}Authenticate with an app{% endtrans %}</h2>
...
-            {{ totp_form.totp_value(autocapitalize="off", autocomplete="off", inputmode="numeric", pattern="[0-9]*", required="required", spellcheck="false", class_="form-group__field", **{"aria-describedby":"totp-errors"}) }}
+            {{ totp_form.totp_value(
+              autocapitalize="off",
+              autocomplete="off",
+              inputmode="numeric",
+              pattern="[0-9]*",
+              required="required",
+              spellcheck="false",
+              class_="form-group__field",
+              aria_describedby="totp-errors",
```
Comment (ewjoachim) [inline warehouse/templates/accounts/two-factor.html:100]: https://wtforms.readthedocs.io/en/2.3.x/widgets/#wtforms.widgets.html_params specifies that `aria_` params are rendered as `aria-` so no need for ** shenanigans.

File: warehouse/templates/accounts/two-factor.html [hunk 2]
```diff
@@ -89,7 +89,17 @@ <h2>{% trans %}Authenticate with an app{% endtrans %}</h2>
...
+              autofocus=not has_webauthn,
```
Comment (ewjoachim) [inline warehouse/templates/accounts/two-factor.html:101]: https://wtforms.readthedocs.io/en/2.3.x/widgets/#wtforms.widgets.html_params specifies that boolean params are handled correctly (Fals = not rendered, True = rendered as `=""`

File: warehouse/templates/accounts/two-factor.html (hunk 1 above)
Comment (ewjoachim) [reply warehouse/templates/accounts/two-factor.html:100]: Capture d’écran 2021-09-03 à 00 03 12
To dismiss any remaining doubt

**Comment by ewjoachim** [comment]
Ok, it took a month for a 10 line PR but testing this was surprisingly annoying :D

For the record, I openned a psql shell:
```
postgres=# \c warehouse
warehouse=# SELECT id FROM users WHERE username = 'ewdurbin';
                  id
--------------------------------------
 70713e05-0f9c-4f30-8899-3c4e8d55e77c
(1 row)

-- simulate webauthn
warehouse=# INSERT INTO "user_security_keys" ("user_id", "credential_id", "label") VALUES ('70713e05-0f9c-4f30-8899-3c4e8d55e77c', '70713e05-0f9c-4f30-8899-3c4e8d55e77c', 'yay');
INSERT 0 1

-- simulate totp
warehouse=# UPDATE users set totp_secret='foo' WHERE username = 'ewdurbin';
UPDATE 1

-- reset env
warehouse=# DELETE FROM user_security_keys;
DELETE 1
warehouse=# UPDATE users set totp_secret=NULL WHERE username = 'ewdurbin';
UPDATE 1
```

**Comment by pradyunsg** [comment]
@ewjoachim Could you rebase this?

**Comment by ewjoachim** [comment]
Done :) (sorry it took some time, my docker setup was broken locally)

**Comment by pradyunsg** [comment]
Thank you, and no worries! I learnt recently that GitHub Codespaces works really well for Warehouse development (after a ~8 minute wait for the build+initdb steps).

I guess we wait for @di or some other committer to review this now. :)

**Pull request merged**